src/Monitor/Tracing/Zipkin.hs

```diff
@@ -174,6 +174,14 @@ tag key val = addSpanEntry (publicKeyPrefix <> key) (tagTextValue val)
 addTag :: Text -> Text -> Builder -> Builder
 addTag key val bldr = bldr { builderTags = Map.insert key (JSON.toJSON val) (builderTags bldr) }
 
+-- | Adds a producer kind tag to a builder. This is a convenience method to use with 'rootSpanWith', for example:
+--
+-- > rootSpanWith addProducerTag alwaysSampled "root" $ action
+--
+-- Please use this method only if you need to create a root producer span.
+addProducerTag :: Builder -> Builder
```

Comment:
Can you rename to `addProducerKind` to avoid further overloading the term "tag" in this file?